src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -45,77 +43,59 @@ internal sealed class InlineHintsTagger : ITagger<IntraTextAdornmentTag>, IDispo
 
-            _tagAggregator = tagAggregator;
             _formatMap = taggerProvider.ClassificationFormatMapService.GetClassificationFormatMap(textView);
             _hintClassification = taggerProvider.ClassificationTypeRegistryService.GetClassificationType(InlineHintsTag.TagId);
+
             _formatMap.ClassificationFormatMappingChanged += this.OnClassificationFormatMappingChanged;
-            _tagAggregator.BatchedTagsChanged += TagAggregator_BatchedTagsChanged;
+            _taggerProvider.GlobalOptionService.AddOptionChangedHandler(this, OnGlobalOptionChanged);
         }
 
-        /// <summary>
-        /// Goes through all the spans in which tags have changed and
-        /// invokes a TagsChanged event. Using the BatchedTagsChangedEvent since it is raised
-        /// on the same thread that created the tag aggregator, unlike TagsChanged.
-        /// </summary>
-        private void TagAggregator_BatchedTagsChanged(object sender, BatchedTagsChangedEventArgs e)
+        public override void Dispose()
         {
-            _taggerProvider.ThreadingContext.ThrowIfNotOnUIThread();
-            InvalidateCache();
-
-            var tagsChanged = TagsChanged;
-            if (tagsChanged is null)
-            {
-                return;
-            }
-
-            var mappingSpans = e.Spans;
-            foreach (var item in mappingSpans)
-            {
-                var spans = item.GetSpans(_buffer);
-                foreach (var span in spans)
-                {
-                    if (tagsChanged != null)
-                    {
-                        tagsChanged.Invoke(this, new SnapshotSpanEventArgs(span));
-                    }
-                }
-            }
+            _formatMap.ClassificationFormatMappingChanged -= OnClassificationFormatMappingChanged;
+            _taggerProvider.GlobalOptionService.RemoveOptionChangedHandler(this, OnGlobalOptionChanged);
+            _underlyingTagger.TagsChanged -= OnTagsChanged;
+            _underlyingTagger.Dispose();
         }
 
         private void OnClassificationFormatMappingChanged(object sender, EventArgs e)
         {
             _taggerProvider.ThreadingContext.ThrowIfNotOnUIThread();
+
+            // When classifications change we need to rebuild the inline tags with updated Font and Color information.
+
             if (_format != null)
             {
                 _format = null;
-                InvalidateCache();
-
-                // When classifications change we need to rebuild the inline tags with updated Font and Color information.
-                var tags = GetTags(new NormalizedSnapshotSpanCollection(_textView.TextViewLines.FormattedSpan));
-
-                foreach (var tag in tags)
-                {
-                    TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(tag.Span));
-                }
+                OnTagsChanged(this, new SnapshotSpanEventArgs(_subjectBuffer.CurrentSnapshot.GetFullSpan()));
```

Comment:
much simpler.  when the classification format changes, we just invalidate everything.